### ISSUE TITLE
Likely wrong precedence of the 'as' operator 

### ISSUE THREAD

**Issue by ksa-real** [jqlang/jq#2446]
**Describe the bug**

In the example below the numbers are squared while the output is expected to be the same as input

```
$ jq '.[]|.*3 as $v|.' <<<'[1, 4, 5,3]'
1
16
25
9
```

Similarly input is doubled but should be the same:
```
$ jq '.[]|.+1 as $v|.' <<<'[1, 4, 5,3]'
2
8
10
6
```

**Environment (please complete the following information):**
 - OS and Version: all platforms including https://jqplay.org/
 - jq version : 1.6

**Comment by wader** [comment]
Not sure if correct but i think the query gets parsed as:
```
$ jq '.[] | (. + (1 as $tt | .))' <<<'[1, 4, 5, 3]'
2
8
10
6
```
which can be simplified to `.[] | (. + (.))` -> `.[] | (. + .)` -> `.[] | .+.`

**Comment by ksa-real** [comment]
Wow. Thanks, indeed this is very likely. Do you know if precedence is specified somewhere in the docs or in some file? I haven't found it.
I wasn't able to understand the precedence looking into https://github.com/stedolan/jq/blob/master/src/parser.y

**Comment by ksa-real** [comment]
I'll go ahead and rename the issue as this is likely a precedence bug.

**Comment by itchyny** [comment]
Fixing the precedence to what you expected will break existing scripts so I'm afraid it is unlikely to be fixed.

**Comment by ksa-real** [comment]
In that case, this needs to be documented. Better, provide a migration path for script authors to fix it, e.g. warning on usage. I'd assume that with the current implementation the intent to use `(.*3) as $v` without parenthesis is very rare and unexpected. Both, mentioned [issue](https://github.com/stedolan/jq/issues/2425) with precedence table and https://github.com/stedolan/jq/wiki/jq-Language-Description#operators-priority doesn't mention it at all. However, there are mentions of constructs like `expr as $NAME | ...` which IMO suggest that binding is a higher-level concept and therefore has lower precedence. So not recognizing `.+1` as an expression in `. + 1 as $v` is very unexpected and every sane engineer understanding this bug probably already put the parenthesis around `.+1` already or in rare cases around `1 as $v`.

**Comment by itchyny** [comment]
I remember that this not precedence problem, the parser only accepts `Term` for binding expression.
https://github.com/stedolan/jq/blob/cff5336ec71b6fee396a95bb0e4bea365e0cd1e8/src/parser.y#L348

**Comment by ksa-real** [comment]
> I remember that this not precedence problem, the parser only accepts `Term` for binding expression.
> 
> https://github.com/stedolan/jq/blob/cff5336ec71b6fee396a95bb0e4bea365e0cd1e8/src/parser.y#L348

Right! The [doc](https://stedolan.github.io/jq/manual/#:~:text=Still%2C%20sometimes%20they,array_length%20%7C%20add%20/%20%24array_length) I referred to:
```
So, there's generally a cleaner way to solve most problems in jq than defining variables. Still, sometimes they do make things easier, so jq lets you define variables using expression as $variable. All variable names start with $. Here's a slightly uglier version of the array-averaging example:

length as $array_length | add / $array_length
```
So, doc says it `as` expects expression on the left, not term.
